02_lecture/homework/task1.ts

```diff
@@ -0,0 +1,11 @@
+async function runSequentially<Type>(
+  array: Array<Type>,
+  func: Function
+): Promise<Type[]> {
+  const resultArray: Array<Type> = [];
+  for (let index in array) {
+    let item = await func.apply(this, [array[index], index]);
+    resultArray.push(item);
+  }
+  return resultArray;
+}
```

Comment:
I've noticed that you often use .call and .apply methods. It isn't necessary to call them every time you want to call a callback. You can call them directly.
For this function I suggest you to add second dynamic type for callback response and replace Function with it
```suggestion
async function runSequentially<Type, TResult>(
  array: Array<Type>,
  func: (element: Type) => Promise<TResult>
): Promise<TResult[]> {
  const resultArray: Array<TResult> = [];
  for (let index in array) {
    let item = await func([array[index], index]);
    resultArray.push(item);
  }
  return resultArray;
}
```

Reply:
done